python/lsst/daf/butler/core/location.py

```diff
@@ -54,6 +55,81 @@ def path(self):
         """
         return os.path.join(self._datastoreRoot, self._uri.path.lstrip("/"))
 
+    @property
+    def component_path(self):
```

Comment:
because I really want to write python code that looks like python code. I'm assuming even though daf_butler is new we aren't allowed to switch because of the "don't confuse lsst_apps people" exemption. I'll switch whilst grumbling.